contracts/SafeDrop.vy

```diff
@@ -0,0 +1,29 @@
+# @version 0.3.10
+
+merkle_root: public(bytes32)
+
+event ClaimMinted:
+  receiver: address
+  claim_amount: uint256
+
+@external
+def __init__(merkle_root: bytes32):
+    self.merkle_root = merkle_root
+
+@external
+def mint(proof: DynArray[bytes32, max_value(uint16)], claim_amount: uint256):
+    leaf: bytes32 = keccak256(concat(convert(msg.sender, bytes32), convert(claim_amount, bytes32)))
+    assert self._process_proof(proof, leaf) == self.merkle_root
+    log ClaimMinted(msg.sender, claim_amount)
+
+@internal
+@pure
+def _process_proof(proof: DynArray[bytes32, max_value(uint16)], leaf: bytes32) -> bytes32:
```

Comment:
for some reason, this always seems to run out of gas
also the proof array is bigger than id expect... i add 8 addresses to the tree all with different claim amount and i proof the first one, i get a list of 160 hashes.